### PR TITLE
Support complex topologies of bridge relays

### PR DESCRIPTION
**Pull request by JonathanLennox**
This allows bridges to be arranged in a tree of full meshes, rather than in a single full-mesh, as currently.  Media packets received from one relay mesh are forwarded to other ones.

Each connection to another relay is identified by a `mesh-id` value, indicating the mesh of relays of which that connection is a part.  It is responsibility of the entity setting up connections between relays (e.g., jicofo) to ensure the topologies are set up correctly.  (This code is not yet written.)

Depends on https://github.com/jitsi/jitsi-xmpp-extensions/pull/70.

### CODE REVIEW

File: jvb/src/main/java/org/jitsi/videobridge/Conference.java [hunk 1]
```diff
@@ -417,6 +417,36 @@ public void sendMessage(
...
+
+        for (Relay relay: relaysById.values())
+        {
+            if (!Objects.equals(meshId, relay.getMeshId()))
+            {
+                relay.sendMessage(msg);
+            }
+        }
+
+        if (tentacle != null && meshId != null)
```
Comment (bgrozev) [inline jvb/src/main/java/org/jitsi/videobridge/Conference.java:444]: This condition is `true` only if we use old octo with mesh IDs?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt [hunk 2]
```diff
@@ -108,6 +108,7 @@ class Relay @JvmOverloads constructor(
     /**
      * True if the ICE agent for this [Relay] will be initialized to serve as a controlling ICE agent, false otherwise.
      */
+    val meshId: String?,
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/Relay.kt:111]: The `meshId` param is between `iceControlling` and its docs. 

Maybe add a note on the semantics of mesh ID somewhere (that it's associated with a bridge-to-bridge link and not a particular bridge)?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt [hunk 3]
```diff
@@ -195,6 +195,8 @@ class RelayMessageTransport(
 
         ep.setVideoType(message.videoType)
 
+        relay.conference.sendMessageFromRelay(message, relay.meshId)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt:198]: This broadcasts the received video type to all local endpoints. Is this the intention?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt [hunk 4]
```diff
@@ -218,6 +220,8 @@ class RelayMessageTransport(
 
         ep.setVideoType(message.sourceName, message.videoType)
 
+        relay.conference.sendMessageFromRelay(message, relay.meshId)
```
Comment (bgrozev) [inline jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt:223]: Same as above

File: jvb/src/main/java/org/jitsi/videobridge/Conference.java (hunk 1 above)
Comment (JonathanLennox) [reply jvb/src/main/java/org/jitsi/videobridge/Conference.java:444]: Yeah, which is impossible, but I was trying to be complete...do you think I should just have a comment saying it cannot happen for old octo?

File: jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt (hunk 3 above)
Comment (JonathanLennox) [reply jvb/src/main/kotlin/org/jitsi/videobridge/relay/RelayMessageTransport.kt:198]: No, will fix.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**